src/cli/commands/install.js

```diff
@@ -232,7 +230,7 @@ export class Install {
       Object.assign(this.resolutions, json.resolutions);
       Object.assign(manifest, json);
 
-      const pushDeps = (depType, {hint, visibility, optional}) => {
+      const pushDeps = (depType, {hint, optional}, isUsed) => {
```

Comment:
I don't know your stance/thoughts on default parameters, but you could default `isUsed = true` here and in the one instance below just pass in the evaluation of production flag.